### DISCUSSION TITLE
Unable to find all instances of a word with page.search_for()

### DISCUSSION THREAD

**Discussion by espitiaandres** [pymupdf/PyMuPDF#1608] [Q&A] [answered]
Hi,

I am currently trying to use the `search_for()` method off `page` to locate where different words appear in a pdf I am trying to search. However, I noticed that when I try to search for a smaller word such as `Commercial Auto`, I am not able to get all instances of the word. For example, what returns from search_for is an empty array indicating that nothing was found. However, in the document I am trying to read I have 3 occurrences of "Commercial Auto". Why is this happening?

> Results of `search_for()` function
<img width="382" alt="image" src="https://user-images.githubusercontent.com/48799891/155222015-92a86198-02d8-481f-a58e-6f510fd6c89d.png">

Here is a snippet of the code I currently have: 
```
# Scrape data for each table, perform logic
def table_scraping(table_name):
  doc_pages = table_scraping.doc_pages
  pdf_bytes = table_scraping.pdf_bytes
  left_margin = table_scraping.left_margin
  total_num_pages = table_scraping.total_num_pages

  for page in doc_pages:
    text_instances = page.search_for(table_name)
    # text_instances_continued = page.search_for(f"Property - Additional Coverages (continued)")
    # TODO: search for (continued) in table names

    if text_instances:
      # tables[table_name] = text_instances
      # continue
      # The origin is at the top of the page, with units in px
      # tables[table_name] = text_instances
      # break
      table_name_instances = list(filter(lambda instance: (instance[0] == left_margin), text_instances))
      # table_name_instance = table_name_instances[0]

      tables[table_name] = {
        # "left_marg": left_margin,
        "table_name_instance": table_name_instances,
      }
      break
```

Please let me know if there is any other information you need me to provide. Thank you!

**Comment by JorjMcKie** [comment]
For classifying it as a bug, mandatory data are missing - primarily the file in question.
So let me reclassify as a question and transfer the issue to the `Discussions` tab.

**Comment by JorjMcKie** [comment, marked as answer]
Could be several explanations, e.g. you may be looking at images instead of at some text, it could be drawings that look like text, etc.
Impossible to say with the file.
As in intermediate track down help, make a sorted list of all words on the page: `page.get_text("words", sort=True)`. This list has tuple items looking like `(x0, y0, x1, y1, "word", block_n, line_n, word_n)`. The first 4 tuple items are rectangle coordinates of the word.
Should give you more clou what is going on.

**Reply by espitiaandres** [reply]
Thank you for your quick response! The `page.get_text("words", sort=True)` solution helped and I noticed that I was doing an indexing mistake in my for loop, which fixed my error. Thank you for the suggestion.
